event_response_tool/ap_creator/create_ap.py

```diff
@@ -97,33 +97,59 @@ def upload_simulation(
                 )
             return simulation
 
-    def build_simulation_data(self):
-        alert.debug("Building simulation data")
+    def build_weighted_simulation_data(self):
+        alert.debug("Building weighted simulation data")
+
         event_column = find_column(
             "event", self.event_weights_df.columns.tolist()
         )
         weight_column = find_column(
             "weight", self.event_weights_df.columns.tolist()
         )
-
         events = self.event_weights_df[event_column].to_list()
         weights = self.event_weights_df[weight_column].to_list()
+
         max_trials = self.trial_count
         trial_count_by_event = [ceil(max_trials * w) for w in weights]
         trial_id = 1
 
         simulation_data = """TrialId,EventId,Day\n"""
 
         for event_id, trial_count in zip(events, trial_count_by_event):
-            if event_id < self.total_num_of_events:
-                for _ in range(0, trial_count):
-                    simulation_data += "%s,%s,%s\n" % (trial_id, event_id, 1)
-                    trial_id += 1
+            for _ in range(0, trial_count):
+                simulation_data += "%s,%s,%s\n" % (trial_id, event_id, 1)
+                trial_id += 1
 
         return simulation_data, sum(trial_count_by_event)
 
-    def create_simulation(self, catalogs, start_date):
-        simulation_data, trial_count = self.build_simulation_data()
+    def build_simulation_data(self):
+        alert.debug("Building simulation data")
+
+        events = [i for i in range(1, self.total_num_of_events + 1)]
+        trial_id = 1
+
+        simulation_data = """TrialId,EventId,Day\n"""
+        max_trials = self.trial_count
+
+        for event_id in events:
+            for _ in range(0, max_trials):
```

Comment:
I timed the execution of building simulation data and it is actually pretty fast since it is just string concatenation. Since it is RDS simulation, I don't expect to see a large number of trials per event (I would say it would mostly be below 500). Below are the results of my test:

1000 trial per event - 233 ms
10000 trial per event - 2312 ms